hedera-node/hedera-mono-service/src/main/java/com/hedera/services/store/contracts/precompile/proxy/ViewExecutor.java

```diff
@@ -120,7 +125,10 @@ private Bytes answerGiven(final int selector) {
             case ABI_ID_GET_NON_FUNGIBLE_TOKEN_INFO -> {
                 final var wrapper =
                         NonFungibleTokenInfoPrecompile.decodeGetNonFungibleTokenInfo(input);
-                final var tokenInfo = stateView.infoForToken(wrapper.tokenID()).orElse(null);
+                final var tokenInfo =
+                        ledgers.infoForToken(
+                                        wrapper.tokenID(), stateView.getNetworkInfo().ledgerId())
+                                .orElse(null);
 
                 validateTrueOrRevert(tokenInfo != null, ResponseCodeEnum.INVALID_TOKEN_ID);
```

Comment:
We can use the new `exists` method mentioned below, instead of fetching `tokenInfo` for token existence validation.